specs/commandline/plugin.md

```diff
@@ -36,15 +36,55 @@ Aliases:
   list, ls
 ```
 
+### notation plugin install
+
+```text
+Installs a plugin
+
+Usage:
+  notation plugin install [--name <plugin name>] [flags] <plugin package> 
+
+Flags:
+  -n, --name string             name of the plugin
+  -p, --plugin-package string   path to the plugin package
+  -h, --help                    help for install
+
+Aliases:
+  install, add
+```
+
+### notation plugin remove 
+
+```text
+Removes a plugin
+
+Usage:
+  notation plugin remove [flags] <plugin name>
+
+Flags:
+  -h, --help          help for remove
+
+Aliases:
+  remove, rm, uninstall, delete
+```
+
 ## Usage
 
 ### Install a plugin
 
-Currently there is no subcommand available for plugin installation. Plugin publisher should provide instructions to download and install the plugin.
+```shell
+notation plugin install --name <plugin name> --plugin-package <plugin package>
+```
+
+Unpon successful execution, the plugins directory is created under the default plugin directory if it does not exist, and the plugin package is extracted to the plugins directory. The plugin installation is then verified. If the verification fails, the plugin is removed from the plugins directory and an error is returned. 
```

Comment:
What are the package formats we are planning to support? .zip?

Reply:
I was thinking `.zip`, `.tar.gz`, `.exe` and the non-extension binary for Darwin and Linux. If the plugin is compressed, only the binary would be extracted and moved to the plugin directory. What are some of the sharp edges you've run into before? I have a prototype working that extracts zip and tar, but I'm certain I haven't tested all the use cases. :)

Reply:
Since `.zip` and `.tar.gz` specification are ambiguous/open, their implementations often suffers from security issues( see [cves](https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=unzip)), here are some the high level issues that I can think of
1. What if there are more than one files(multiple plugins, unrelated files, etc) inside zip, should we copy everything? 
2. What if zip is a [zip bomb](https://en.wikipedia.org/wiki/Zip_bomb) ? 
3. What if unzipping tries to write outside the designated pugin folder ([zip slip](https://security.snyk.io/research/zip-slip-vulnerability))?
4. Should we support symlinks in zip, (see [symlink-vulnerability]( https://effortlesssecurity.in/2020/08/05/zip-symlink-vulnerability/))?

Reply:
I see, it's not as simple as it appears. Starting with the .exe and binary for Darwin and Linux sounds good to me. :)

Reply:
@priteshbandi I can answer a few.

> What if there are more than one files(multiple plugins, unrelated files, etc) inside zip, should we copy everything?

Yes, we actually should. It may contain license files, config files for global configs, and other plugin assets.

> What if zip is a [zip bomb](https://en.wikipedia.org/wiki/Zip_bomb) ?

We can limit the max plugin size.

> What if unzipping tries to write outside the designated pugin folder ([zip slip](https://security.snyk.io/research/zip-slip-vulnerability))?

We can restrict that. Actually, `oras` does the same restriction.

> Should we support symlinks in zip, (see [symlink-vulnerability](https://effortlesssecurity.in/2020/08/05/zip-symlink-vulnerability/))?

No, we should not. Symlinks and hard links cause more problem than benefits. Actually, we should support extracting regular files within a flat folder only.